std_msgs/UInt8 bin_id
std_msgs/String location